### PR TITLE
VL-122_Typescript-errors-fix_Vitalii-Dudnik

### PR DESCRIPTION
**Pull request by KinduD21**
Fix TS errors in different component groups

https://ilevel.atlassian.net/jira/software/c/projects/VL/boards/11?selectedIssue=VL-122

### CODE REVIEW

File: src/ui.navigation-pagination/UPagination.vue [hunk 1]
```diff
@@ -108,24 +109,24 @@ const pageButtons = computed(() => {
 });
 
 const prevIsDisabled = computed(() => {
-  return props.disabled || currentPage.value === null || currentPage.value <= 1;
+  return props.disabled || currentPage.value == null || currentPage.value <= 1;
```
Comment (itsJohnnyGrid) [inline src/ui.navigation-pagination/UPagination.vue:112]: == ??? why ?

File: src/ui.navigation-pagination/UPagination.vue [hunk 2]
```diff
@@ -108,24 +109,24 @@ const pageButtons = computed(() => {
...
 });
 
 const nextIsDisabled = computed(() => {
-  return props.disabled || currentPage.value === null || currentPage.value >= totalPages.value;
+  return props.disabled || currentPage.value == null || currentPage.value >= totalPages.value;
```
Comment (itsJohnnyGrid) [inline src/ui.navigation-pagination/UPagination.vue:116]: same

File: src/ui.navigation-pagination/UPagination.vue [hunk 3]
```diff
@@ -108,24 +109,24 @@ const pageButtons = computed(() => {
...
 });
 
 function selectPage(page: number) {
   currentPage.value = page;
 }
 
 function goToPrevPage() {
-  currentPage.value = currentPage.value === null ? 1 : Math.max(currentPage.value - 1, 1);
+  currentPage.value = currentPage.value == null ? 1 : Math.max(currentPage.value - 1, 1);
```
Comment (itsJohnnyGrid) [inline src/ui.navigation-pagination/UPagination.vue:124]: same

File: src/ui.navigation-pagination/UPagination.vue [hunk 4]
```diff
@@ -108,24 +109,24 @@ const pageButtons = computed(() => {
...
 }
 
 function goToNextPage() {
   currentPage.value =
-    currentPage.value === null
+    currentPage.value == null
```
Comment (itsJohnnyGrid) [inline src/ui.navigation-pagination/UPagination.vue:129]: same

File: src/ui.text-files/UFiles.vue [hunk 5]
```diff
@@ -39,7 +46,7 @@ const formattedFileList = computed(() =>
   }),
 );
 
-function onRemoveFile(fileId) {
+function onRemoveFile(fileId: number) {
```
Comment (itsJohnnyGrid) [inline src/ui.text-files/UFiles.vue:49]: may be string also

File: src/ui.text-notify/UNotify.vue [hunk 6]
```diff
@@ -105,7 +114,9 @@ function setPosition() {
   if (props.xPosition === POSITION.center) {
     styles.left = `calc(50% - ${notifyWidth / 2}px)`;
   } else {
-    styles[props.xPosition] = "0px";
+    const xPos = props.xPosition as "left" | "right";
```
Comment (itsJohnnyGrid) [inline src/ui.text-notify/UNotify.vue:117]: `props.xPosition` is typed? this looks weird to me

File: src/ui.container-page/UPage.vue [hunk 7]
```diff
@@ -69,11 +70,16 @@ const isShownArrowButton = computed(() => {
...
-  const classes = props.gray
-    ? config.value.htmlBody.split(" ").filter((item) => Boolean(item))
-    : "";
+  const classes =
+    props.gray && config.value?.htmlBody
+      ? config.value.htmlBody.split(" ").filter((item) => Boolean(item))
+      : "";
 
-  document.querySelector("body").classList.add(...classes);
+  const body = document.querySelector("body");
```
Comment (Explicit12) [inline src/ui.container-page/UPage.vue:78]: There is no need to use `querySelector` here. Just use `document.body`.

File: src/ui.text-files/types.ts [hunk 8]
```diff
@@ -1,12 +1,14 @@
...
+import type { UFileItem } from "../ui.text-file/types.ts";
+
 export type Config = Partial<typeof defaultConfig>;
 
 export interface UFilesProps {
   /**
    * List of file objects.
    */
-  fileList?: Array<object>;
+  fileList?: UFileItem[];
```
Comment (Explicit12) [inline src/ui.text-files/types.ts:11]: It should accept array of [File objects](https://developer.mozilla.org/en-US/docs/Web/API/File)

File: src/ui.text-files/UFiles.vue [hunk 9]
```diff
@@ -57,11 +64,11 @@ function onRemoveFile(fileId) {
...
-          :id="file.id"
-          :key="file.id"
-          :label="file.label"
-          :url="file.url"
-          :image-url="file.imageUrl"
+          :id="file?.id"
+          :key="file?.id"
+          :label="file?.label"
+          :url="file?.url"
+          :image-url="file?.imageUrl"
```
Comment (Explicit12) [inline src/ui.text-files/UFiles.vue:71]: Replace UFileItem with File and you probably don't need those optional chaining operators

**Review by Explicit12 — CHANGES_REQUESTED** [review]

**Review by Explicit12 — APPROVED** [review]

File: src/ui.text-files/UFiles.vue [hunk 10]
```diff
@@ -29,7 +29,14 @@ const emit = defineEmits([
 const { filesLabelAttrs, itemsAttrs, itemAttrs } = useAttrs(props);
 
 const formattedFileList = computed(() =>
-  props.fileList.map((file) => {
+  props.fileList?.map((file) => {
+    if (!(file instanceof File)) {
```
Comment (itsJohnnyGrid) [inline src/ui.text-files/UFiles.vue:33]: may be Blob also I guess.

File: src/ui.text-files/types.ts [hunk 11]
```diff
@@ -6,7 +6,7 @@ export interface UFilesProps {
   /**
    * List of file objects.
    */
-  fileList?: Array<object>;
+  fileList?: File[] | FileList;
```
Comment (itsJohnnyGrid) [inline src/ui.text-files/types.ts:9]: Blob ?

File: src/ui.text-files/UFiles.vue [hunk 12]
```diff
@@ -43,8 +43,8 @@ const formattedFileList = computed(() => {
       }
 
       return {
-        id: file instanceof File ? file.name : "unknown",
-        label: file instanceof File ? file.name : "unknown",
+        id: file instanceof File ? file.name : file instanceof Blob ? file.type : "",
+        label: file instanceof File ? file.name : file instanceof Blob ? file.type : "",
```
Comment (itsJohnnyGrid) [inline src/ui.text-files/UFiles.vue:47]: Are you sure? need to discuss.

File: src/ui.text-files/UFiles.vue [hunk 13]
```diff
@@ -42,9 +42,12 @@ const formattedFileList = computed(() => {
         return null;
```
Comment (itsJohnnyGrid) [inline src/ui.text-files/UFiles.vue:42]: need to discuss

**Review by itsJohnnyGrid — APPROVED** [review]

**Pull request merged**